components/generator/imagevector/src/test/kotlin/io/github/composegears/valkyrie/generator/imagevector/lazyproperty/SvgGradientParserTest.kt

```diff
@@ -0,0 +1,370 @@
+package io.github.composegears.valkyrie.generator.imagevector.lazyproperty
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import io.github.composegears.valkyrie.generator.imagevector.ImageVectorGenerator
+import io.github.composegears.valkyrie.generator.imagevector.ImageVectorGeneratorConfig
+import io.github.composegears.valkyrie.generator.imagevector.OutputFormat.LazyDelegateProperty
+import io.github.composegears.valkyrie.generator.imagevector.backingproperty.loadIcon
+import io.github.composegears.valkyrie.parser.IconParser
+import org.junit.jupiter.api.Test
+
+class SvgGradientParserTest {
+
+    @Test
+    fun `svg linear gradient parsing`() {
```

Comment:
You can test these with params using `@ParameterizedTest`.

Reply:
Like

```kt
    @ParameterizedTest
    @EnumSource(value = OutputFormat::class)
    fun `svg linear gradient parsing`(outputFormat: OutputFormat) {
        val icon = loadIcon("ic_linear_gradient.svg")
        val parserOutput = IconParser.toVector(icon)
        val output = ImageVectorGenerator.convert(
            vector = parserOutput.vector,
            kotlinName = parserOutput.kotlinName,
            config = ImageVectorGeneratorConfig(
                packageName = "io.github.composegears.valkyrie.icons",
                packName = "",
                nestedPackName = "",
                outputFormat = outputFormat,
                generatePreview = false,
            ),
        ).content

        val expected = when(outputFormat) {
            OutputFormat.BackingProperty -> Path("resources/1.kt").readText()
            OutputFormat.LazyDelegateProperty -> Path("resources/2.kt").readText()
        }
        assertThat(output).isEqualTo(expected)
    }
```

Reply:
Yes, make sense together with this suggestion https://github.com/ComposeGears/Valkyrie/pull/70#discussion_r1695038212

Reply:
Tests now so clean, thanks for this suggestion!